plugins/utils/dressing.nix

```
@@ -0,0 +1,428 @@
+          "[20 0.2]"
+          ''
+            Min width of window.
+
+            Can be a list of mixed types, e.g. `[140 0.9]` means "less than 140 columns or 90% of
+            total."
+          '';
+
+        buf_options = helpers.defaultNullOpts.mkAttrsOf types.anything "{}" ''
+          An attribute set of neovim buffer options.
+        '';
+
+        win_options =
+          helpers.defaultNullOpts.mkAttrsOf types.anything
+          ''
+            {
+              wrap = false;
+              list = true;
+              listchars = "precedes:...,extends:...";
+              sidescrolloff = 0;
+            }
+          ''
+          "An attribute set of window options.";
+
+        mappings =
+          helpers.defaultNullOpts.mkAttrsOf (with types; attrsOf (either str (enum [false])))
+          ''
+            {
+              n = {
+                "<Esc>" = "Close";
+                "<CR>" = "Confirm";
+              };
+              i = {
+                "<C-c>" = "Close";
+                "<CR>" = "Confirm";
+                "<Up>" = "HistoryPrev";
+                "<Down>" = "HistoryNext";
+              };
+            }
+          ''
+          ''
+            Mappings for defined modes.
+
+            To disable a default mapping in a specific mode, set it to `false`.
+          '';
+
+        override =
+          helpers.defaultNullOpts.mkLuaFn
+          ''
+            "function(conf) return conf end"
+          ''
+          ''
+            Lua function that takes config that is passed to nvim_open_win.
+            Used to customize the layout.
+          '';
+
+        get_config = helpers.defaultNullOpts.mkLuaFn "null" ''
+          This can be a function that accepts the opts parameter that is
+          passed in to 'vim.select' or 'vim.input'. It must return either
+          nil or config values to use in place of the global config values
+          for that module.
+
+          See :help dressing_get_config for more info.
+        '';
+      };
+
+      select = {
+        enabled = helpers.defaultNullOpts.mkBool true ''
+          Enable the vim.ui.select implementation.
+        '';
+
+        backend =
+          helpers.defaultNullOpts.mkListOf types.str
+          ''
+            ["telescope" "fzf_lua" "fzf" "builtin" "nui"]
+          ''
+          "Priority list of preferred vim.select implementations. ";
+
+        trim_prompt = helpers.defaultNullOpts.mkBool true ''
+          Trim trailing `:` from prompt.
+        '';
+
+        telescope =
+          helpers.defaultNullOpts.mkNullable
+          (with helpers.nixvimTypes; either strLua (attrsOf anything)) "null"
+          ''
+            Options for telescope selector.
+
+            Can be a raw lua string like:
+
+            `telescope = \'\'require("telescope.themes").get_ivy({})\'\'`
+
+            or an attribute set of telescope settings.
+          '';
+
+        fzf.window =
```

```suggestion
        fzf =
```
Just let `fzf` be an option.

---

Done.